Makefile

```diff
@@ -99,3 +133,57 @@ verify-shellscripts-initramfs:
 		--severity error \
 		defaults/linuxrc \
 		defaults/initrd.scripts
+
+$(BUILD_DIR)/build-config:
+# $(addprefix $(BUILD_DIR)/temp/,$(TEMPFILES))
+	install -d $(BUILD_DIR)
+	echo ${PREFIX} > $(BUILD_DIR)/PREFIX
+	echo ${BINDIR} > $(BUILD_DIR)/BINDIR
+	echo ${SYSCONFDIR} > $(BUILD_DIR)/SYSCONFDIR
+	echo ${MANDIR} > $(BUILD_DIR)/MANDIR
+	touch $(BUILD_DIR)/build-config
+
+$(BUILD_DIR)/software.sh:
+	install -d $(BUILD_DIR)/temp/
+	echo -e $(SOFTWARE_VERSION) > $(BUILD_DIR)/temp/versions
+	cat $(BUILD_DIR)/temp/versions defaults/software.sh > $(BUILD_DIR)/software.sh
+
+$(BUILD_DIR)/doc/genkernel.8.txt:
+	install -d $(BUILD_DIR)/doc/
+	cp doc/genkernel.8.txt $(BUILD_DIR)/doc/genkernel.8.txt
+
+$(BUILD_DIR)/%: %
+	install -d $(BUILD_DIR)/
+	cp $< $@
+
+$(BUILD_DIR)/genkernel: $(addprefix $(BUILD_DIR)/,$(FINAL_DEPS)) $(BUILD_DIR)/software.sh
+	cp genkernel $(BUILD_DIR)/genkernel
+
+install: PREFIX := $(file <$(BUILD_DIR)/PREFIX)
+install: BINDIR := $(file <$(BUILD_DIR)/BINDIR)
+install: SYSCONFDIR := $(file <$(BUILD_DIR)/SYSCONFDIR)
+install: MANDIR := $(file <$(BUILD_DIR)/MANDIR)
```

Comment:
is there a timing issue here? when are these rules evaluated? can you make sure it's always after the files are created? maybe throw error checking here to ensure they are non-empty.

Reply:
addressed in commit [ee3045f](https://github.com/gentoo/genkernel/pull/56/commits/ee3045fe2dca73560e685d535d9a14220bee7f20)